Update model release checklist (#1724)
huggingface/hub-docs: docs/hub/model-release-checklist.md

Bonus: a recognised library also allows you to track downloads of your model over time.

2. **Correct Metadata**:
   - **Pipeline Tag:** Choose the correct [pipeline tag](https://huggingface.co/docs/hub/model-cards#specifying-a-task--pipelinetag-) that accurately reflects your model's primary task. This tag determines how your model appears in search results and which widgets are displayed on your model page.

   Examples of common pipeline tags:
   - `text-generation` - For language models that generate text
   - `text-to-image` - For text-to-image generation models
   - `image-text-to-text` - For vision-language models (VLMs) that generate text
   - `text-to-speech` - For models that generate audio from text
  

>    - **License:** Properly filling the license information in metadata will help users see if they can use the model for commercial or research purposes.

Reviewer: ```suggestion
   - **License:** License information is crucial for users to understand how they can use the model.
```

   
   Try this model directly in your browser: [Space Demo](https://huggingface.co/spaces/username/model-demo)
   ```
   

> When you write a demo, please download the model from its repository in the Hub (instead of using external sources) so both are cross-linked together. This will help all your artifacts achieve greater visibility.

Reviewer 2: ```suggestion
When you create a demo, please download the model from its repository on the Hub (instead of using external sources like Google Drive); it cross-links model artefacts and demo together and allows more paths to visibility. 
```

unifying it with the line above.

Reviewer: yeah, looks better, thanks!

> It's a good practise to load the model from the model's own repository in the demo instead of loading them from drive or keeping the checkpoint in the repository. This way, Space is linked to the repository, and chances of your model making it to trending is higher due to increased downloads.

Reviewer: ```suggestion
When you write a demo, please download the model from its repository in the Hub (instead of using external sources) so both are cross-linked together. This will help all your artifacts achieve greater visibility.
```
